### PR TITLE
Support streaming in composable behaviors

### LINKED ISSUE #902: Allow actors based on composable behaviors to use the streaming API
Actors based on composable behaviors store a `typed_actor_view`, which currently does not allow actors to call `make_sink` etc.

### PR DESCRIPTION
**Pull request by Neverlord**
Close #902.

### CODE REVIEW

File: libcaf_core/caf/composed_type.hpp [hunk 1]
```diff
@@ -58,82 +57,36 @@ struct composed_type<detail::type_list<>, Ys, Zs, detail::type_list<Rs...>> {
 // end of inner loop Ys (Zs)
 template <class X, class... Xs, class Ys, class Rs>
 struct composed_type<detail::type_list<X, Xs...>, Ys, detail::type_list<>, Rs>
-    : composed_type<detail::type_list<Xs...>, Ys, Ys, Rs> {};
+  : composed_type<detail::type_list<Xs...>, Ys, Ys, Rs> {};
 
 // case #1
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libcaf_core/caf/composed_type.hpp]: I cleaned up the Doxygen docu as well as the comments. It's messy template metaprogramming, but I hope it's at least a biter easier to comprehend now.

File: libcaf_core/caf/attach_continuous_stream_stage.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  using state_type = typename Trait::state;
+  static_assert(std::is_same<
+                  void(state_type&),
+                  typename detail::get_callable_trait<Init>::fun_sig>::value,
+                "Expected signature `void (State&)` for init function");
+  static_assert(std::is_same<
+                  void(state_type&, downstream<output_type>&, input_type),
+                  typename detail::get_callable_trait<Fun>::fun_sig>::value,
+                "Expected signature `void (State&, downstream<Out>&, In)` "
+                "for consume function");
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libcaf_core/caf/attach_continuous_stream_stage.hpp]: I'm not sure what you are asking about. Which types are you referring to?

File: libcaf_core/caf/attach_continuous_stream_source.hpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#include "caf/fwd.hpp"
+#include "caf/policy/arg.hpp"
+#include "caf/stream_source.hpp"
+#include "caf/stream_source_driver.hpp"
+#include "caf/stream_source_trait.hpp"
+
+namespace caf {
+
+/// Creates a new continuous stream source by instantiating the default
+/// source implementation with `Driver. `The returned manager is not
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libcaf_core/caf/attach_continuous_stream_source.hpp]: We're usually using verbatim formatting to highlight (template) parameters, e.g., http://actor-framework.org/doc/classcaf_1_1scheduled__actor.html#a0dcc12450ea5cd15ca482fb1974b1453. Would you recommend a different style?

File: examples/streaming/integer_stream.cpp [hunk 4]
```diff
@@ -2,7 +2,7 @@
  *     Basic, non-interactive streaming example for processing integers.      *
  ******************************************************************************/
 
-// Manual refs: lines 17-52, 54-87, 89-120, 133-139 (Streaming)
+// Manual refs: lines 17-46, 48-78, 80-107, 121-125 (Streaming)
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply examples/streaming/integer_stream.cpp]: Excellent catch! 👍

File: libcaf_core/caf/attach_continuous_stream_source.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (josephnoir) [reply libcaf_core/caf/attach_continuous_stream_source.hpp]: The comment reads as follows:

    with `Driver. `The

... but should probably be:

    with `Driver`. The

It's just a the order of the tick, dot, and space I was commenting on. I could have expressed that better, sorry.

File: libcaf_core/caf/attach_continuous_stream_stage.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (josephnoir) [reply libcaf_core/caf/attach_continuous_stream_stage.hpp]: The function `attach_continuous_stream_source`, [here](https://github.com/actor-framework/actor-framework/blob/b653ff07fed9c60ec2a9317d2a1a3f71465f8c19/libcaf_core/caf/attach_continuous_stream_source.hpp#L66) has similar template parameters, but do not have similar asserts to check the types.

There are a few more, [attach_stream_sink](https://github.com/actor-framework/actor-framework/blob/issue/902/libcaf_core/caf/attach_stream_sink.hpp#L55), [attach_stream_source](https://github.com/actor-framework/actor-framework/blob/b653ff07fed9c60ec2a9317d2a1a3f71465f8c19/libcaf_core/caf/attach_stream_source.hpp#L153), and another [attach_stream_source](https://github.com/actor-framework/actor-framework/blob/b653ff07fed9c60ec2a9317d2a1a3f71465f8c19/libcaf_core/caf/attach_stream_source.hpp#L97).

I'm not sure if the information to implement the checks would be available.

**Pull request merged**